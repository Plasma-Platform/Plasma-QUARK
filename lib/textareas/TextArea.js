'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _N2B = require('ui-toolkit/lib/notifications/N2B');

var _N2B2 = _interopRequireDefault(_N2B);

var _TextAreaStateless = require('./TextAreaStateless.js');

var _TextAreaStateless2 = _interopRequireDefault(_TextAreaStateless);

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TextArea = function (_Component) {
  (0, _inherits3.default)(TextArea, _Component);

  function TextArea(props, context) {
    (0, _classCallCheck3.default)(this, TextArea);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TextArea.__proto__ || (0, _getPrototypeOf2.default)(TextArea)).call(this, props, context));

    _this.getValue = function () {
      return _this.state.value;
    };

    _this.componentWillReceiveProps = function (nextProps) {
      if (nextProps.value !== _this.props.value) {
        _this.setState({
          value: nextProps.value,
          fieldFilled: true
        });
      }
    };

    _this.setValidationStatus = function (status, notificationText) {
      _this.handleValidation({ status: status, notificationText: notificationText });
    };

    _this.onChange = function (e) {
      if (typeof _this.props.onChange === 'function') {
        _this.props.onChange(e);
      }

      _this.setState({
        value: _this.comp.input.target.textareaInput.value
      });

      _this.zeroingInputCounter();
    };

    _this.onFocus = function (e) {
      _this.setState({
        isValid: null,
        notificationText: '',
        fieldFilled: true,
        inFocus: true
      });
    };

    _this.onBlur = function (callback, e) {
      // обрезаем пробелы по краям введенного значения
      if (_this.state.value) {
        _this.setState({
          value: _this.state.value.trim()
        });
      }

      if (!_this.state.value) {
        _this.setState({
          fieldFilled: false
        });
      }

      // сбрасываем статус, что курсор юзера находится в поле
      _this.setState({
        inFocus: false
      });

      if (typeof _this.props.onValidate === 'function') {
        _this.props.onValidate().then(function (data) {
          _this.handleValidation(data);
          callback();
        });
      }
    };

    _this.handleValidation = function (data) {
      _this.setState({
        isValid: data.status,
        notificationText: data.message
      });
    };

    _this.zeroingInputCounter = function () {
      var textareaElement = _this.comp.input.target.textareaInput;

      if (textareaElement.value === 0) {
        return;
      }

      var currentValue = _this.props.maxLength - (textareaElement ? textareaElement.value.length : 0),
          currentMaxValue = Math.max(0, currentValue);

      _this.setState({
        limitCounter: currentMaxValue
      });
    };

    _this.component = _this.component || (0, _utils.connectNotificationTextField)(_TextAreaStateless2.default);


    _this.state = {
      value: props.value || '',
      limitCounter: 0,
      fieldFilled: !!props.value,
      inFocus: false,
      isValid: null,
      notificationtext: _this.props.notificationText || ''
    };
    return _this;
  }

  // приводим к нулю, возможные отрицательные значения, которые появляются в счетчике
  // при монтировании компонента с заполненным значением, превышающим установвленный лимит


  (0, _createClass3.default)(TextArea, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var DecoratedTextAreaField = this.component;

      return _react2.default.createElement(DecoratedTextAreaField, {
        ref: function ref(comp) {
          return _this2.comp = comp;
        },
        id: this.props.id,
        fieldType: this.props.fieldType,
        type: this.props.type,
        label: this.props.label,
        className: this.props.className,
        disabled: this.props.disabled,
        onChange: this.onChange,
        maxLength: this.props.maxLength,
        limitCounter: this.state.limitCounter,
        value: this.state.value,
        isValid: this.state.isValid,
        fieldFilled: this.state.fieldFilled,
        onFocus: this.onFocus,
        inFocus: this.state.inFocus,
        onBlur: this.onBlur,
        notification: { code: 'N2B', text: this.state.notificationText }
      });
    }
  }]);
  return TextArea;
}(_react.Component);

exports.default = TextArea;