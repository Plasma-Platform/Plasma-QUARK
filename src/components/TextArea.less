@import "../assets/styles/variables";

.text-area {
  display: block;
  padding-top: 20px;
  padding-left: 20px;
  padding-bottom: 20px;
  transition: .3s;
  overflow: hidden;

  & &__input {
    display: block;
    overflow: auto;
    resize: none;
    border: none;
    background: none;
    padding-right: calc(~'50px + 17px');
    width: calc(~'100% + 17px');
    padding-left: 0;
    min-height: 125px;
  }

  & &__label {
    position: absolute;
    top: 19px;
    left: 20px;
    transition: font-size .3s;
  }

  & &__notification-icon {
    opacity: 0;
    top: 20px;
  }

  &_filled {
    padding-top: 40px;
  }
  
  &_focused {
    background: #fff;
    border-color: #90caf9;
    box-shadow: none;
  }

  &_disabled {
    background: #fff;
    border-color: #eceff1;
    box-shadow: none;
    color: #eceff1;
    cursor: not-allowed;
    user-select: none;

    [class*="__label"] {
      color: #eceff1;
      user-select: none;
    }

    .text-area__limit {
      color: #eceff1;
    }
  }

  &_ta2,
  &_ta4,
  &_ta6,
  &_ta8 {
    padding-left: 50px;

    .text-area__type-icon,
    .text-area__notification-icon {
      top: 21px;
    }

    .text-area__label {
      left: 52px;
    }
  }

  &_ta3,
  &_ta4,
  &_ta7,
  &_ta8 {
    .text-area__input,
    .text-area__label {
      font-size: 16px;
    }
  }

  &_valid,
  &_invalid {
    .text-area__notification-icon {
      opacity: 1;
    }
  }

  &.animated {
    animation-timing-function: @easeInOutCubic ;
    animation-delay: .15s, .75s;
    animation-duration: .25s;
    animation-fill-mode: both;
  }

  &_valid.animated {
    animation-name: showSuccessField, hideSuccessField;
  }

  &_invalid.animated {
    animation-name: showErrorField, hideErrorField;
  }

  &__limit {
    bottom: 10px;
    color: @secodary;
    font-size: 11px;
    font-style: normal;
    font-weight: normal;
    position: absolute;
    right: 20px;
    text-align: right;
    z-index: 50;
  }

  &_ta7,
  &_ta8 {
    .text-area__limit {
      font-size: 13px;
    }
  }
}