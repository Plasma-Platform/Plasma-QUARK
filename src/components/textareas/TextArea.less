@import "../../../styles/variables";

.TextArea {
  resize: none;
  display: block;

  &_wrapper {
    position: relative;

    &:after {
      position: absolute;
      content: '';
      display: block;
      width: ~'calc(100% - 45px)';
      height: @spacing4x;
      left: 1px;
      top: 1px;
      z-index: 5;
      background: @light-grey;
      border-radius: 3px 0 0 0;
      box-shadow: inset 0 1px 0 0 @light-grey-2;
      transition: .3s;
    }

    &.TextArea_inFocus {
      &:after {
        box-shadow: none;
        background: @color-white;
      }
    }
  }

  &__input {
    width: 100%;
    min-height: 110px;
    height: auto;
    max-height: 274px;
    background: @light-grey;
    box-shadow: inset 0 1px 0 0 @light-grey-2;
    color: @basic;
    border: 1px solid transparent;
    border-radius: 3px;
    outline: none;
    padding: @spacing4x @spacing2x @spacing2x @spacing2x;
    font-size: 14px;
    transition: .3s;

    &::placeholder {
      color: @secodary;
    }

    &:focus {
      background: @color-white;
      box-shadow: none;
      border-color: @light-blue;

      &::placeholder {
        color: @light-blue-2;
      }
    }

    &:disabled {
      border-color: @light-grey;
      color: @light-grey-2;
      background: @color-white;
      box-shadow: none;
      cursor: not-allowed;
      user-select: none;

      &::placeholder,
      & ~ .TextArea__label {
        color: @light-grey-2;
        user-select: none;
      }
    }
  }

  &__label {
    position: absolute;
    z-index: 6;
    top: 20px;
    left: 20px;
    color: @secodary;
    cursor: text;
    transition: .3s;
    font-size: 14px;
  }

  &__type-icon,
  &__notification-icon {
    position: absolute !important;
    width: 20px;
    height: 20px;
    z-index: 200;
  }

  &__type-icon {
    top: 11px;
    left: 20px;
  }

  &__notification-icon {
    right: 20px;
    top: 20px;
    opacity: 0;
    transition: opacity .25s @easeInOutCubic;
  }

  &_fieldFilled {
    .TextArea__label {
      font-size: 11px !important;
    }
  }

  &_TA2, &_TA4, &_TA6, &_TA8 {
    .TextArea__type-icon,
    .TextArea__notification-icon {
      top: 21px;
    }

    .TextArea__input {
      padding-left: 50px;
    }

    .TextArea__label {
      padding-left: 30px;
    }
  }

  &_TA3, &_TA4, &_TA7, &_TA8 {
    .TextArea__input {
      font-size: 16px;
    }

    .TextArea__label {
      font-size: 16px;
    }
  }

  &_valid, &_invalid {
    .TextArea__notification-icon {
      opacity: 1;
    }
  }

  &_valid {
    .TextArea__input {
      animation: showSuccessField .25s @easeInOutCubic forwards, hideSuccessField .25s @easeInOutCubic .75s both;
    }

    &:after {
      animation: showSuccessField .25s @easeInOutCubic forwards, hideSuccessField .25s @easeInOutCubic .75s both;
    }
  }
  &_invalid {
    .TextArea__input {
      animation: showErrorField .25s @easeInOutCubic forwards, hideErrorField .25s @easeInOutCubic .75s both;
    }
    &:after {
      animation: showErrorField .25s @easeInOutCubic forwards, hideErrorField .25s @easeInOutCubic .75s both;
    }

  }

  &__limit {
    font-size: 11px;
    font-weight: normal;
    font-style: normal;
    text-align: right;
    color: @secodary;
    position: absolute;
    z-index: 50;
    right: 10px;
    bottom: 10px;
  }

  &_TA7, &_TA8 {
    .TextArea__limit {
      font-size: 13px;
    }
  }
}

@keyframes showSuccessField {
  0% {
    background: @light-grey;
    box-shadow: inset 0 1px 0 0 @light-grey-2;
  }
  100% {
    background: @light-green;
    box-shadow: inset 0 1px 0 0 @light-green-2;
  }
}

@keyframes hideSuccessField {
  0% {
    background: @light-green;
    box-shadow: inset 0 1px 0 0 @light-green-2;
  }
  100% {
    background: @light-grey;
    box-shadow: inset 0 1px 0 0 @light-grey-2;
  }
}

@keyframes showErrorField {
  0% {
    background: @light-grey;
    box-shadow: inset 0 1px 0 0 @light-grey-2;
  }
  100% {
    background: @warning-light;
    box-shadow: inset 0 1px 0 0 @warning;
  }
}

@keyframes hideErrorField {
  0% {
    background: @warning-light;
    box-shadow: inset 0 1px 0 0 @warning;
  }
  100% {
    background: @light-grey;
    box-shadow: inset 0 1px 0 0 @light-grey-2;
  }
}


