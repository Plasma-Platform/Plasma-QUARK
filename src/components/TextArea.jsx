import React, {Component, PropTypes}  from 'react';
import classnames from 'classnames';
import Textarea from 'react-textarea-autosize';

import './AbstractField.less';
import './TextArea.less';

export default class TextArea extends Component {

  static propTypes = {
    id           : PropTypes.string,
    sizeType     : PropTypes.oneOf(['TA1', 'TA2', 'TA3', 'TA4', 'TA5', 'TA6', 'TA7', 'TA8']).isRequired,
    placeholder  : PropTypes.string,
    className    : PropTypes.string,
    value        : PropTypes.string,
    disabled     : PropTypes.bool,
    filled       : PropTypes.bool,
    animated     : PropTypes.bool,
    limitCounter : PropTypes.number,
    onBlur       : PropTypes.func,
    onFocus      : PropTypes.func,
    onChange     : PropTypes.func,
    onValidate   : PropTypes.func,
    label        : PropTypes.string,
    customIcon   : PropTypes.string
  };

  render () {
    const textAreaWrapClassname = classnames(`abstract-field text-area text-area_${this.props.sizeType.toLowerCase()}`, this.props.className, {
      'text-area_filled'   : this.props.filled,
      'text-area_focused'  : this.props.focused,
      'text-area_valid'    : this.props.isValid,
      'text-area_disabled' : this.props.disabled,
      'text-area_invalid'  : (this.props.isValid !== null) && !this.props.isValid,
      'animated'           : this.props.animated
    });

    const textAreaClassname = 'text-area__input';
    const labelClassname = 'text-area__label';
    const customIcon = this.props.customIcon;
    const iconClassname = classnames('text-area__type-icon', 'icon', {
      'icon-letter'    : !customIcon && this.props.type === 'email',
      'icon-magnifier' : !customIcon && this.props.type === 'search',
      'icon-key'       : !customIcon && this.props.type === 'password',
      [customIcon]     : customIcon
    });

    const iconNotificationClassname = classnames('text-area__notification-icon', 'icon', {
      'icon-alert'        : this.props.isValid !== null && !this.props.isValid,
      'icon-check_circle' : this.props.isValid
    });

    return (
      <div
        className={textAreaWrapClassname}
        onFocus={this.props.onFocus}
        onBlur={this.props.onBlur}
      >
        {['TA2', 'TA4', 'TA6', 'TA8'].indexOf(this.props.sizeType) >= 0 &&
        <span className={ iconClassname }/>
        }

        <Textarea
          ref={ textareaInput => this.input = textareaInput }
          id={this.props.id}
          className={textAreaClassname}
          disabled={this.props.disabled}
          onChange={this.props.onChange}
          maxLength={this.props.maxLength}
          minRows={3}
          maxRows={15}
          value={this.props.value}
        />

        <label
          className={labelClassname}
          htmlFor={this.props.id}
        >
          { this.props.label || this.props.placeholder}
        </label>

        <span className={ iconNotificationClassname }/>

        {this.props.maxLength
          ? <div className={`text-area__limit
                        ${!this.props.disabled && (this.props.value || this.props.filled)
          ? 'text-area__limit--active'
          : ''}`}
        >
          {this.props.limitCounter}
        </div>
          : null
        }
      </div>
    );
  }
}
