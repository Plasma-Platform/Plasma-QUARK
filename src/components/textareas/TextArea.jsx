import React, {Component, PropTypes}  from 'react';
import N2B                            from 'ui-toolkit/lib/notifications/N2B';
import TextareaStateless              from './TextAreaStateless.jsx';
import {connectNotificationTextField} from '../utils';

import './TextArea.less';

export default class TextArea extends Component {

  constructor(props, context) {
    super(props, context);

    this.state = {
      value: props.value || '',
      limitCounter: 0,
      fieldFilled: !!props.value,
      inFocus: false,
      isValid: null,
      notificationtext: this.props.notificationText || ''
    }
  }

  getValue = () => {
    return this.state.value;
  };

  componentWillReceiveProps = (nextProps) => {
    if (nextProps.value !== this.props.value) {
      this.setState({
        value: nextProps.value,
        fieldFilled: true
      });
    }
  }

  setValidationStatus = (status, notificationText) => {
    this.handleValidation({status, notificationText});
  }

  onChange = (e) => {
    if (typeof this.props.onChange === 'function') {
      this.props.onChange(e);
    }

    this.setState({
      value: this.comp.input.target.textareaInput.value
    });

    this.zeroingInputCounter();
  }

  onFocus = (e) => {
    this.setState({
      isValid: null,
      notificationText: '',
      fieldFilled: true,
      inFocus: true
    });
  }

  onBlur = (callback, e) => {
    // обрезаем пробелы по краям введенного значения
    if (this.state.value) {
      this.setState({
        value: this.state.value.trim()
      });
    }

    if (!this.state.value) {
      this.setState({
        fieldFilled: false
      });
    }

    // сбрасываем статус, что курсор юзера находится в поле
    this.setState({
      inFocus: false
    });

    if (typeof this.props.onValidate === 'function') {
      this.props.onValidate().then((data) => {
        this.handleValidation(data);
        callback();
      });
    }
  }

  handleValidation = (data) => {
    this.setState({
      isValid: data.status,
      notificationText: data.message
    });
  }

  // приводим к нулю, возможные отрицательные значения, которые появляются в счетчике
  // при монтировании компонента с заполненным значением, превышающим установвленный лимит
  zeroingInputCounter = () => {
    let textareaElement = this.comp.input.target.textareaInput;

    if (textareaElement.value === 0) {
      return;
    }

    let currentValue = this.props.maxLength - (textareaElement ? textareaElement.value.length : 0),
      currentMaxValue = Math.max(0, currentValue);

    this.setState({
      limitCounter: currentMaxValue
    });
  }

  component = this.component || connectNotificationTextField(TextareaStateless);

  render() {
    const DecoratedTextAreaField = this.component;

    return (
      <DecoratedTextAreaField
        ref={ comp => this.comp = comp }
        id={this.props.id}
        fieldType={this.props.fieldType}
        type={this.props.type}
        label={this.props.label}
        className={this.props.className}
        disabled={this.props.disabled}
        onChange={this.onChange}
        maxLength={this.props.maxLength}
        limitCounter={this.state.limitCounter}
        value={this.state.value}
        isValid={this.state.isValid}
        fieldFilled={this.state.fieldFilled}
        onFocus={this.onFocus}
        inFocus={this.state.inFocus}
        onBlur={this.onBlur}
        notification={ {code: 'N2B', text: this.state.notificationText} }
      />
    );
  }
}
