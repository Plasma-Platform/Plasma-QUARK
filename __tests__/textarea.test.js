/* global expect:false, test:false, it:false, describe:false, jest:false, beforeEach:false */

import React from 'react';
import {shallow, mount} from 'enzyme';
import TextArea from '../src/components/textareas';
import TextAreaPresent from '../src/components/TextArea.jsx';

describe('1 TextArea high level components [TA1-8] testing', () => {
  let callback;
  let textAreas = [];

  beforeEach(() => {
    callback = jest.fn();
    let taCount = Object.keys(TextArea).length;
    for (let i = 1; i <= taCount; i++) {
      let componentType = `TA${i}`;
      let component = TextArea[componentType];
      textAreas.push(component);
    }
  });

  describe('1.1 Events testing', () => {
    it('1.1.1 calls onFocus callback', () => {
      textAreas.forEach((Component) => {
        let testNode = shallow(<Component onFocus={callback} />);
        testNode.simulate('focus');
        expect(callback).toBeCalled();
      });
    });

    it('1.1.2 calls onBlur callback', () => {
      textAreas.forEach((Component) => {
        let testNode = shallow(<Component onBlur={callback} />);
        testNode.simulate('blur');
        expect(callback).toBeCalled();
      });
    });

    it('1.1.3 calls onChange callback', () => {
      textAreas.forEach((Component) => {
        let testNode = shallow(<Component onChange={callback} />);
        testNode.simulate('change');
        expect(callback).toBeCalled();
      });
    });
  });

  describe('1.2 API', () => {
    it('1.2.1 performs focus using focus() method', () => {
      textAreas.forEach((Component) => {
        let testNode = mount(<Component />);
        testNode.node.focus();
        let classes = testNode.find('.abstract-field').props().className;
        expect(classes).toMatch('focused');
      });
    });

    it('1.2.2 gets value of textarea', () => {
      textAreas.forEach((Component) => {
        let testNode = mount(<Component value="test text" />);
        let value = testNode.node.getValue();
        expect(value).toMatch('test text');
      });
    });
  });
});

describe('2 Snapshot testing', () => {
  it('2.1 takes a shot of all High level text area types (default props)', () => {
    for (let i = 1; i <= 8; i++) {
      let type = `TA${i}`;
      let TA = TextArea[type];
      let component = shallow(<TA />).html();
      expect(component).toMatchSnapshot();
    }
  });

  it('2.2 takes a shot of disabled stateless TextArea components (TextArea.jsx)', () => {
    let component = shallow(<TextAreaPresent
      value="test"
      disabled
    />).html();
    expect(component).toMatchSnapshot();
  });

  it('2.3 takes a shot of stateless TextArea components (TextArea.jsx)', () => {
    let component = shallow(<TextAreaPresent value="test" />).html();
    expect(component).toMatchSnapshot();
  });
});

describe('3 Presentational TextArea testing', () => {
  let callback;
  beforeEach(() => {
    callback = jest.fn();
  });

  it('3.1 simulates focus using focus() method', () => {
    let component = mount(<TextAreaPresent onFocus={callback} />);
    component.node.focus();
    expect(callback).toBeCalled();
  });
});
