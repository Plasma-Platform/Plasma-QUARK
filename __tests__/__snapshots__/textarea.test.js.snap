exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 1`] = `"<div class=\"text-area text-area_ta1 abstract-field field-style\"><div class=\"text-area__trim\"><textarea class=\"text-area__input\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 2`] = `"<div class=\"text-area text-area_ta2 abstract-field field-style\"><div class=\"text-area__trim\"><span class=\"text-area__type-icon icon icon-magnifier\"></span><textarea class=\"text-area__input\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 3`] = `"<div class=\"text-area text-area_ta3 abstract-field field-style\"><div class=\"text-area__trim\"><textarea class=\"text-area__input\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 4`] = `"<div class=\"text-area text-area_ta4 abstract-field field-style\"><div class=\"text-area__trim\"><span class=\"text-area__type-icon icon icon-magnifier\"></span><textarea class=\"text-area__input\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 5`] = `"<div class=\"text-area text-area_ta5 abstract-field field-style\"><div class=\"text-area__trim\"><textarea class=\"text-area__input\" maxlength=\"2000\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span><span class=\"text-area__limit\">2000</span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 6`] = `"<div class=\"text-area text-area_ta6 abstract-field field-style\"><div class=\"text-area__trim\"><span class=\"text-area__type-icon icon icon-magnifier\"></span><textarea class=\"text-area__input\" maxlength=\"2000\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span><span class=\"text-area__limit\">2000</span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 7`] = `"<div class=\"text-area text-area_ta7 abstract-field field-style\"><div class=\"text-area__trim\"><textarea class=\"text-area__input\" maxlength=\"2000\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span><span class=\"text-area__limit\">2000</span></div></div>"`;

exports[`2 Snapshot testing 2.1 takes a shot of all High level text area types (default props) 8`] = `"<div class=\"text-area text-area_ta8 abstract-field field-style\"><div class=\"text-area__trim\"><span class=\"text-area__type-icon icon icon-magnifier\"></span><textarea class=\"text-area__input\" maxlength=\"2000\" style=\"height:0;\"></textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon\"></span><span class=\"text-area__limit\">2000</span></div></div>"`;

exports[`2 Snapshot testing 2.2 takes a shot of disabled stateless TextArea components (TextArea.jsx) 1`] = `"<div class=\"text-area text-area_ta1 abstract-field field-style text-area_disabled text-area_invalid\"><div class=\"text-area__trim\"><textarea class=\"text-area__input\" disabled=\"\" style=\"height:0;\">test</textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon icon-alert\"></span></div></div>"`;

exports[`2 Snapshot testing 2.3 takes a shot of stateless TextArea components (TextArea.jsx) 1`] = `"<div class=\"text-area text-area_ta1 abstract-field field-style text-area_invalid\"><div class=\"text-area__trim\"><textarea class=\"text-area__input\" style=\"height:0;\">test</textarea><label class=\"text-area__label\"></label><span class=\"text-area__notification-icon icon icon-alert\"></span></div></div>"`;
